### ISSUE TITLE
can't compute cl_00 of a map with healpy nside smaller than 32

### ISSUE THREAD

**Issue by kateyang0** [LSSTDESC/NaMaster#120]
Tried to use the code "example 1" on [https://namaster.readthedocs.io/en/latest/sample_simple.html] to compute cl_00 (TT) of temperature maps with different nside. The code failed to run and gave error "zsh: segmentation fault" (when ran on jupyter notebook the kernel died) with nside = 4, 8 16, but ran successfully at nside=32 and 64. This is confusing as NaMaster should work for any nside.

**Comment by damonge** [comment]
Humm, I'm not able to reproduce this error. Did you install namaster from pip?

**Comment by kateyang0** [comment]
Below is the code I used. The map I used is just random values with different nside. Again it works for nside>=32.
[namaster-example1.txt](https://github.com/LSSTDESC/NaMaster/files/5873808/namaster-example1.txt)

**Comment by fjaviersanchez** [comment]
@damonge, I can reproduce the error on a mac-os based system with a conda install. I'll take a deeper look at it (it breaks in the apodization part already).

**Comment by damonge** [comment]
OK, thanks! I can see how apodization might fail. Can you check if that's the only place it fails?

**Comment by fjaviersanchez** [comment]
Yeah, the rest works. Here's what it says when it breaks (and you set `OMP_NUM_THREADS=1`.

```
//anaconda3/lib/python3.7/site-packages/pymaster/utils.py in mask_apodization(mask_in, aposize, apotype)
    128     """
    129     return lib.apomask(mask_in.astype("float64"), len(mask_in),
--> 130                        aposize, apotype)
    131 
    132 

RuntimeError: Not enough memory in listir
```

**Comment by fjaviersanchez** [comment]
I'm checking the code and I'd say that the error comes from [here](https://github.com/LSSTDESC/NaMaster/blob/master/src/healpix_extra.c#L472) but I don't know under what circumstances it would break with just one thread...

**Comment by fjaviersanchez** [comment]
@kateyang0 you should be able to use the example with Nside=4 if you change the apodization radius to a larger value. For example:
`mask = nmt.mask_apodization(mask, 7.7, apotype="Smooth")` (anything larger than 7.7 should work).

TLDR;
I continued digging and the problem is that the apodization radius is too small for Nside=4 and that generates problems downstream. In particular, with 1 deg the variable `lenlist0` in [here](https://github.com/LSSTDESC/NaMaster/blob/master/src/nmt_mask.c#L45)
takes the value 0. Then, it gets divided by 2 in the for loop that follows (so anything lower than 2 can give potential problems). And even when it's larger than 2, it continues breaking further downstream. I guess that we should probably raise an exception if the apodization radius is "too small". I don't see an easy way around this really. Thoughts @damonge?

**Comment by fjaviersanchez** [comment]
I continued digging and the problem is that the apodization radius is too small for Nside=4 and that generates problems downstream. In particular, with 1 deg the variable `lenlist0` in [here](https://github.com/LSSTDESC/NaMaster/blob/master/src/nmt_mask.c#L45) takes the value 0. Then, it gets divided by 2 in the for loop that follows (so anything lower than 2 can give potential problems). And even when it's larger than 2, it continues breaking further downstream. I guess that we should probably raise an exception if the apodization radius is "too small". I don't see an easy way around this really. Thoughts @damonge?

**Comment by kateyang0** [comment]
@fjaviersanchez What is the best way to estimate the reasonable apodize radius for each nside? Is there a relation between the two that we can use? I tried apodize radius of 7.7 for nside=4 but it still didn't work.I also tried apodize radius of 10 or 20, neither of them worked.  But the code worked if I remove the apodizing line of code and use the original mask. 
Thanks for pointing this parameter out!

**Comment by damonge** [comment]
@fjaviersanchez thanks a lot. This is definitely a problem with the C-level apodizing function and we should just propagate the error to python (which for some reason isn't happening, I'll look into that).

@kateyang0 I'd suggest you work without any apodization. At these super large scales apodization shouldn't really matter (and if it does, you should be using a higher resolution map and mask, where this problem won't appear).

**Comment by damonge** [comment]
@kateyang0 this issue will be closed by #124, but let me know if you keep having trouble
